EngineTest/TestWindow.h

```diff
@@ -0,0 +1,152 @@
+		Atom quit_msg = XInternAtom(display, "QUIT_MSG", false);		
+
+		XEvent xev;
+		// NOTE: we use an if statement here because we are not handling all events in this translation
+		//       unit, so XPending(display) will often not ever be 0, and therefore this can create
+		//       an infinite loop... but this protects XNextEvent from blocking if there are no events.
+		if (XPending(display) > 0)
+		{
+			XNextEvent(display, &xev);
+			switch (xev.type)
+			{
+				case ConfigureNotify:
+				{
+					XConfigureEvent xce{ xev.xconfigure };
+
+					// NOTE: This event is generated for a variety of reasons, so
+					//		 we need to check to see which window generated the event, 
+					//		 and the check if this was a window resize.
+					for (u32 i{ 0 }; i < _countof(_windows); ++i)
+					{
+						if (*((Window*)_windows[i].handle()) == xev.xany.window)
+						{
+							if ((u32)xce.width != _windows[i].width() || (u32)xce.height != _windows[i].height())
+							{
+								_windows[i].resize((u32)xce.width, (u32)xce.height);
+							}
+						}
+					}
+					break;
+				}
+				case ClientMessage:
+					if ((Atom)xev.xclient.data.l[0] == wm_delete_window)
+					{      
+						// Find which window was sent the close event, and call function
+						for (u32 i{ 0 }; i < _countof(_windows); i++)
+						{
+							if (*((Window*)_windows[i].handle()) == xev.xany.window)
+							{
+								_windows[i].close();
+							}
+						}
+
+						// Check if all _windows are closed, and exit application if so
+						bool all_closed{ true };
+						for (u32 i{ 0 }; i < _countof(_windows); i++)
+						{
+							if (!_windows[i].is_closed())
+							{
+								all_closed = false;
+							}
+						}
+						if (all_closed)
+						{
+							// Set up quit message and send it using dummy window
+							XEvent close;
+							close.xclient.type = ClientMessage;
+							close.xclient.serial = window;
+							close.xclient.send_event = true;
+							close.xclient.message_type = XInternAtom(display, "QUIT_MSG", false);
+							close.xclient.format = 32;
+							close.xclient.window = 0;
+							close.xclient.data.l[0] = XInternAtom(display, "QUIT_MSG", false);
+							XSendEvent(display, window, false, NoEventMask, &close);
+						}
+					}
+					else
+					{
+						// Dont handle this here
+                        XPutBackEvent(display, &xev);
+					}
+					break;
+				case KeyPress:
+					// NOTE: "state" represents the keys held down prior to the key press the
+					//		 keycode represents - the numeric evaluation is also different.
+					if (xev.xkey.state == State::ALT && xev.xkey.keycode == Key::ENTER)
+					{
+						for (u32 i{ 0 }; i < _countof(_windows); i++)
+						{
+							if (*((Window*)_windows[i].handle()) == xev.xany.window)
+							{
+								_windows[i].set_fullscreen(!_windows[i].is_fullscreen());
+							}
+						}
+					}
+			}
+		}
+	}
+#endif // _WIN64
+
+// shutdown() function from Primal goes here
```

Comment:
Don't  need shutdown function for Linux to close/shutdown graphics for example? Or does it get handled in `run()` `if(all_closed)` block? If so, maybe make a commented note?

Reply:
The `shutdown()` function for Windows are Linux are identical,  so there should just be one.